src/main/java/com/rackspace/ceres/app/helper/MetricDeletionHelper.java

```diff
@@ -0,0 +1,179 @@
+ * limitations under the License.
+ *
+ */
+
+package com.rackspace.ceres.app.helper;
+
+import com.github.benmanes.caffeine.cache.AsyncCache;
+import com.rackspace.ceres.app.config.AppProperties;
+import com.rackspace.ceres.app.model.SeriesSetCacheKey;
+import com.rackspace.ceres.app.services.DataTablesStatements;
+import java.time.Instant;
+import org.springframework.data.cassandra.core.cql.ReactiveCqlTemplate;
+import org.springframework.data.redis.core.ReactiveStringRedisTemplate;
+import org.springframework.stereotype.Component;
+import reactor.core.publisher.Flux;
+import reactor.core.publisher.Mono;
+
+@Component
+public class MetricDeletionHelper {
+
+  private static final String PREFIX_SERIES_SET_HASHES = "seriesSetHashes";
+  private static final String DELIM = "|";
+
+  private AppProperties appProperties;
+  private final DataTablesStatements dataTablesStatements;
+  private final ReactiveCqlTemplate cqlTemplate;
+  private final AsyncCache<SeriesSetCacheKey, Boolean> seriesSetExistenceCache;
+  private final ReactiveStringRedisTemplate redisTemplate;
+
+  public MetricDeletionHelper(AppProperties appProperties,
+      DataTablesStatements dataTablesStatements,
+      ReactiveCqlTemplate cqlTemplate, ReactiveStringRedisTemplate redisTemplate,
+      AsyncCache<SeriesSetCacheKey, Boolean> seriesSetExistenceCache) {
+    this.appProperties = appProperties;
+    this.dataTablesStatements = dataTablesStatements;
+    this.cqlTemplate = cqlTemplate;
+    this.seriesSetExistenceCache = seriesSetExistenceCache;
+    this.redisTemplate = redisTemplate;
+  }
+
+  /**
+   * Delete metric_names by tenant and metric name.
+   *
+   * @param tenant     the tenant
+   * @param metricName the metric name
+   * @return the mono
+   */
+  public Mono<Boolean> deleteMetricNamesByTenantAndMetricName(String tenant, String metricName) {
+    return cqlTemplate
+        .execute("DELETE FROM metric_names WHERE tenant = ? AND metric_name = ?",
+            tenant, metricName)
+        .retryWhen(appProperties.getRetryDelete().build());
+  }
+
+  /**
+   * Delete series_sets by tenant id and metric name.
+   *
+   * @param tenant     the tenant
+   * @param metricName the metric name
+   * @return the mono
+   */
+  public Mono<Boolean> deleteSeriesSetsByTenantIdAndMetricName(String tenant,
+      String metricName) {
+    return cqlTemplate
+        .execute("DELETE FROM series_sets WHERE tenant = ? AND metric_name = ?"
+            , tenant, metricName)
+        .retryWhen(appProperties.getRetryDelete().build());
+  }
+
+  /**
+   * Delete series_set_hashes by tenant and seriesSetHash.
+   *
+   * @param tenant        the tenant
+   * @param seriesSetHash the series set hash
+   * @return the mono
+   */
+  public Mono<Boolean> deleteSeriesSetHashes(String tenant, String seriesSetHash) {
+    return cqlTemplate
+        .execute("DELETE FROM series_set_hashes WHERE tenant = ? AND series_set_hash = ?",
```

Comment:
Need to move this queries to constants I guess

Reply:
these will be taken care in another PR https://github.com/racker/ceres/pull/43